mailgun/route_handlers.py

```diff
@@ -30,193 +30,222 @@
+        # otherwise, send the email to the list
+        member_addresses = list(member_addresses)
+        logger.debug(u'Full list of recipients: %s', member_addresses)
+
+        # if we found the course instance, insert [SHORT TITLE] into the subject
+        if ci and ci.short_title:
+            title_prefix = '[{}]'.format(ci.short_title)
+            if title_prefix not in subject:
+                subject = title_prefix + ' ' + subject
+
+        # we want to add 'via Canvas' to the sender's name.  so first make
+        # sure we know their name.
+        logger.debug(u'Original sender name: %s, address: %s',
+                     sender_display_name, sender_address)
+        if not sender_display_name:
+            name = get_name_for_email(ml.canvas_course_id, sender_address)
+            if name:
+                sender_display_name = name
+                logger.debug(u'Looked up sender name: %s, address: %s',
+                             sender_display_name, sender_address)
+
+        # now add in 'via Canvas'
+        if sender_display_name:
+            sender_display_name += ' via Canvas'
+        logger.debug(u'Final sender name: %s, address: %s',
+                     sender_display_name, sender_address)
+
+        # make sure inline images actually show up inline, since fscking
+        # mailgun won't let us specify the cid on post.  see their docs at
+        #   https://documentation.mailgun.com/user_manual.html#sending-via-api
+        # where they explain that they use the inlined file's name attribute
+        # as the content-id.
+        if inlines:
+            for f in inlines:
+                logger.debug(u'Replacing "%s" with "%s" in body', f.cid, f.name)
+                body_plain = re.sub(f.cid, f.name, body_plain)
+                body_html = re.sub(f.cid, f.name, body_html)
+
+        # convert the original to/cc fields back to strings so we can send
+        # them along through the listserv
+        original_to_list = [a.full_spec() for a in to_list]
+        original_cc_list = [a.full_spec() for a in cc_list]
+
+        # and send it off
+        logger.debug(
+            u'Mailgun router handler sending email to %s from %s, subject %s',
+            member_addresses, parsed_sender.full_spec(), subject)
+        try:
+            ml.send_mail(
+                sender_display_name, sender_address,
+                member_addresses, subject, text=body_plain, html=body_html,
+                original_to_address=original_to_list, original_cc_address=original_cc_list,
+                attachments=attachments, inlines=inlines, message_id=message_id
+            )
+        except RuntimeError:
+            logger.exception(
+                u'Error attempting to send message from %s to %s, originally '
+                u'sent to list %s, with subject %s', parsed_sender.full_spec(),
+                member_addresses, ml.address, subject)
+            raise JsonResponse({'success': False}, status=500)
```

Comment:
Yeah, there's nothing in the language that requires something you raise to be an exception.  To my mind, it doesn't make sense to define a custom exception class just to embed the response in, and pull it out in the `except` clause.  Especially since this is nominally a private method.